### PR TITLE
Feature/localization singleton

### PR DESCRIPTION
**Pull request by jorre127**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
 # Changelog
+## [9.0.0] - 2022-10-03
+### Breaking
+- Translations can no longer be accessed from static methods on the Localization object. Instead you now need to manually manage the different localization instances. Migration steps are described in the readme.
```
Comment (vanlooverenkoen) [inline CHANGELOG.md:4]: Localization `class`

File: README.md [hunk 2]
```diff
@@ -56,6 +56,30 @@ flutter packages pub run locale_gen
 pub run locale_gen
 ```
 
+### Migration steps <9.0.0 to >=9.0.0
+With version 9.0.0 of local gen you no longer have static access to the translations, instead you can now manually manage the different localization instances. You can for example store a localization instance as a static. This way you can use it largely the same way as before. Example from the example project:
+
+```dart
+class LocaleViewModel with ChangeNotifier {
+  static Localization? localization;
```
Comment (vanlooverenkoen) [inline README.md:64]: static localization = Localization();

File: README.md [hunk 3]
```diff
@@ -56,6 +56,30 @@ flutter packages pub run locale_gen
...
+  ...
+
+  void init(){
+    _localization = Localization()
```
Comment (vanlooverenkoen) [inline README.md:68]: no need to only initialize it here.

File: README.md [hunk 4]
```diff
@@ -56,6 +56,30 @@ flutter packages pub run locale_gen
...
+      localizationOverrides: customLocalizationOverrides,
+    );
+  }
+```
+
+You can then access this localization instance anywhere in the project like:
+```dart
+LocaleViewModel.localization.translation1;
+LocaleViewModel.localization.translation2;
+LocaleViewModel.localization.translation3;
```
Comment (vanlooverenkoen) [inline README.md:80]:
```
yourLocalizationInstance.translation1
yourLocalizationInstance.translation2
yourLocalizationInstance.translation3
```

File: example/lib/app.dart [hunk 5]
```diff
@@ -20,8 +19,8 @@ class MyApp extends StatelessWidget {
             GlobalMaterialLocalizations.delegate,
             GlobalWidgetsLocalizations.delegate,
           ],
-          locale: Localization.locale,
-          supportedLocales: Localization.supportedLocales,
+          locale: LocaleViewModel.localization.locale,
```
Comment (vanlooverenkoen) [inline example/lib/app.dart:22]: I would name localization -> localizationInstance

File: example/lib/viewmodel/locale/locale_viewmodel.dart [hunk 6]
```diff
@@ -6,6 +6,9 @@ import 'package:locale_gen_example/util/locale/localization.dart';
 class LocaleViewModel with ChangeNotifier {
   final LocaleRepository _localeRepository;
   var customLocalizationOverrides = CustomLocalizationOverrideManager();
+  static Localization? _localization;
+
+  static Localization get localization => _localization ?? Localization();
```
Comment (vanlooverenkoen) [inline example/lib/viewmodel/locale/locale_viewmodel.dart:11]: no need to make it optional and use a getter. This can be just 1 line code

**Review by vanlooverenkoen — CHANGES_REQUESTED** [review]

**Review by vanlooverenkoen — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -56,6 +56,29 @@ flutter packages pub run locale_gen
...
+```dart
+class LocaleViewModel with ChangeNotifier {
+  static final Localization localizationInstance = Localization();
+  ...
+
+  void init(){
+    await Localization.load(
+      locale: locale,
+      localizationOverrides: customLocalizationOverrides,
+    );
```
Comment (ikbendewilliam) [inline README.md:71]: notifyListeners?

File: README.md [hunk 8]
```diff
@@ -56,6 +56,29 @@ flutter packages pub run locale_gen
...
+### Migration steps <9.0.0 to >=9.0.0
+With version 9.0.0 of local gen you no longer have static access to the translations, instead you can now manually manage the different localization instances. You can for example store a localization instance as a static. This way you can use it largely the same way as before. Example from the example project:
+
+```dart
+class LocaleViewModel with ChangeNotifier {
+  static final Localization localizationInstance = Localization();
+  ...
+
+  void init(){
+    await Localization.load(
```
Comment (ikbendewilliam) [inline README.md:68]: We can't await in `void init()` since it's not a Future

**Review by ikbendewilliam — APPROVED** [review]

**Pull request merged**